# Computational Bioscience & Machine Learning 2018/2019 - KAUST
## Practical Session: Genome Assembly

### Structure

1. Quality control on input data 
2. Assembling a small genome sequence
3. Assembly QC
4. Annotating genes 
5. Using long reads for assembly
6. Hybrid Assembly QC


### Software

#### QC
* [FASTQC](http://www.bioinformatics.babraham.ac.uk/projects/fastqc)
* [K-mer analysis toolkit](https://kat.readthedocs.io)
* [Trimmomatic](https://github.com/timflutre/trimmomatic)
* [MultiQC](https://multiqc.info)
* [Kraken2](https://ccb.jhu.edu/software/kraken2/index.shtml?t=manual)
* [Quality Assessment Tool for Genome Assemblies](http://quast.sourceforge.net/quast)

#### Assembly
* [SPADes](http://cab.spbu.ru/software/spades/)

#### Annotation
* [PROKKA](https://github.com/tseemann/prokka)

#### HPC Environment â€“ Job Scheduling
* [Slurm](https://slurm.schedmd.com/)
